pkg/operator/encryption/controllers/migration_controller.go

```diff
@@ -103,46 +104,65 @@ func NewMigrationController(
 		apiServerConfigInformer.Informer(), // do not remove, used by the precondition checker
 		deployer,
 	).ToController(
-		c.name, // don't change what is passed here unless you also remove the old FooDegraded condition
+		c.controllerInstanceName,
 		eventRecorder.WithComponentSuffix("encryption-migration-controller"),
 	)
 }
 
 func (c *migrationController) sync(ctx context.Context, syncCtx factory.SyncContext) (err error) {
-	degradedCondition := &operatorv1.OperatorCondition{Type: "EncryptionMigrationControllerDegraded", Status: operatorv1.ConditionFalse}
-	progressingCondition := &operatorv1.OperatorCondition{Type: "EncryptionMigrationControllerProgressing", Status: operatorv1.ConditionFalse}
+	// Status for these conditions is left out to make sure it's correctly set in every branch
+	degradedCondition := applyoperatorv1.OperatorCondition().
+		WithType("EncryptionMigrationControllerDegraded")
+	progressingCondition := applyoperatorv1.OperatorCondition().
+		WithType("EncryptionMigrationControllerProgressing")
+
 	defer func() {
 		if degradedCondition == nil && progressingCondition == nil {
 			return
 		}
```

Comment:
if one is nil and not the other, fail.

Reply:
Makes sense, if one of these is nil, _WithConditions_ will panic, so I made this an OR instead